ds248x/ds248x.go

```diff
@@ -154,6 +161,70 @@ func (d *Dev) Search(alarmOnly bool) ([]onewire.Address, error) {
 	return onewire.Search(d, alarmOnly)
 }
 
+// ChannelSelect function is for selecting one of eight 1-w channels on DS2482-800. 
+// On other chips it does nothing. Function silently limits channel selection between
+// 0 and 7. It is expected that application keeps track of
+// with 1-w device is connected to with channel.
+//
+// Communication error is returned if present.
+//
+func (d *Dev) ChannelSelect(ch int) (err error) {
+	switch d.isDS248x {
+	case isDS2482x100:
+
+	case isDS2482x800:
+		if ch < 0 {
+			ch = 0
+		}
+		if ch > 7 {
+			ch = 7
+		}
+		csc := []byte{cscIO0w, cscIO1w, cscIO2w, cscIO3w, cscIO4w, cscIO5w, cscIO6w, cscIO7w}
+		buf := []byte{cmdChannelSelect, csc[ch]}
+		if err = d.i2c.Tx(buf, nil); err != nil {
+			return fmt.Errorf("ds2482-800: error while selecting channel: %s", err)
+		}
+	case isDS2483:
+
+	default:
+
+	}
+	return
+}
+
+// SelectedChannel function is to read with 1-w channel selected on DS2482-800. 
+// On other chips it always returns 0. It is expected that application keeps track of
+// with 1-w device is connected to with channel.
+//
+// On error returns 255.
```

Comment:
Why 255? It seems a bit random since it's a int, not a uint8.

Reply:
From 8 bit time... Could be more elegant by I wanted only one variable in return.

Reply:
One last change: I would prefer to change it to -1? This would remove an hardcoded value (255) that has no immediate meaning to the user. -1 is more often used as a sentinel value and is easier to check, like `if ch < 0 {`